src/averageAges.js

```diff
@@ -36,8 +41,15 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const resultPeople = people.filter(person =>
+    person.sex === 'f' && (withChildren ? people.some(human =>
+      human.mother === person.name) : true));
```

Comment:
```suggestion
    person.sex === 'f' && (
    withChildren 
      ? people.some(human => human.mother === person.name) 
      : true)
    );
```
pay attention to the code formatting